managed/services/management/service.go

```diff
@@ -144,6 +146,96 @@ func (s *ServiceService) RemoveService(ctx context.Context, req *managementpb.Re
 	return &managementpb.RemoveServiceResponse{}, nil
 }
 
+// AddCustomLabels adds or replaces (if key exists) custom labels for a service.
+func (s *ServiceService) AddCustomLabels(ctx context.Context, req *managementpb.AddCustomLabelsRequest) (*managementpb.AddCustomLabelsResponse, error) {
+	if req.ServiceId == "" {
+		return nil, status.Error(codes.InvalidArgument, "service_id is required")
+	}
```

Comment:
This check can be moved to the protobuf level using the `string_not_empty` field validator

Reply:
I didn't see the validators being used very often.
Is the expected design to use validators instead of custom validation in the service?

Reply:
Not sure about the company-wide conventions, but it's definitely being used in the portal APIs. 
I think it makes the codebase cleaner, however it's up to you to decide what option to use.

Reply:
Sounds great. Thank you